apps/api/src/modules/fixtures/fixtures.controller.ts

```diff
@@ -0,0 +1,198 @@
+      squadId: number;
+      opponent: string;
+      kickoffAt: string;
+      actingCoachId: number;
+      location?: string;
+      notes?: string;
+    } = {
+      squadId,
+      opponent: body.opponent,
+      kickoffAt: body.kickoffAt,
+      actingCoachId: user.id,
+    };
+
+    if (body.location) {
+      payload.location = body.location;
+    }
+    if (body.notes) {
+      payload.notes = body.notes;
+    }
+
+    const fixture = await fixturesService.createFixture(payload);
+
+    res.status(201).json(fixture);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function getFixture(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const user = (req as any).user as ReqUser | undefined;
+    if (!user) {
+      return next(Object.assign(new Error("Unauthorized"), { status: 401 }));
+    }
+
+    const { id: fixtureId } = fixtureIdParamSchema.parse(req.params);
+    const fixture = await fixturesService.getFixtureById(fixtureId);
+
+    if (user.role === "COACH") {
+      const owns = await pool.query(
+        `SELECT id FROM squads WHERE id=$1 AND coach_id=$2`,
+        [fixture.squadId, user.id]
+      );
+      if ((owns.rowCount ?? 0) === 0) {
+        return next(Object.assign(new Error("Forbidden"), { status: 403 }));
+      }
+    } else {
+      const member = await pool.query(
+        `SELECT id FROM squad_members WHERE squad_id=$1 AND user_id=$2`,
+        [fixture.squadId, user.id]
+      );
+      if ((member.rowCount ?? 0) === 0) {
+        return next(Object.assign(new Error("Forbidden"), { status: 403 }));
+      }
+    }
+
+    res.json(fixture);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function listFixturesForSquad(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const user = (req as any).user as ReqUser | undefined;
+    if (!user) {
+      return next(Object.assign(new Error("Unauthorized"), { status: 401 }));
+    }
+
+    const { id: squadId } = squadIdParamSchema.parse(req.params);
+
+    if (user.role === "COACH") {
+      const owns = await pool.query(
+        `SELECT id FROM squads WHERE id=$1 AND coach_id=$2`,
+        [squadId, user.id]
+      );
+      if ((owns.rowCount ?? 0) === 0) {
+        return next(Object.assign(new Error("Forbidden"), { status: 403 }));
+      }
+    } else {
+      const member = await pool.query(
+        `SELECT id FROM squad_members WHERE squad_id=$1 AND user_id=$2`,
+        [squadId, user.id]
+      );
+      if ((member.rowCount ?? 0) === 0) {
+        return next(Object.assign(new Error("Forbidden"), { status: 403 }));
+      }
+    }
+
+    const fixtures = await fixturesService.listFixturesForSquad(squadId);
+    res.json(fixtures);
+  } catch (error) {
+    next(error);
+  }
+}
+
+export async function setAvailability(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const user = (req as any).user as ReqUser | undefined;
+    if (!user) {
```

Comment:
Right now, both createFixture and getFixture start with this auth check. This is boilerplate that can/should be moved to your middleware layer.

If you handle auth in middleware, you can:
- Guarantee that req.user always exists when the controller runs.
- Remove duplication and keep controllers cleaner.
- Centralize error handling (401s, expired sessions, etc.) in one place.